operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ExecutorServiceManager.java

```diff
@@ -17,6 +18,8 @@ public class ExecutorServiceManager {
   private static ExecutorServiceManager instance;
   private final ExecutorService executor;
   private final ExecutorService workflowExecutor;
+  private final ForkJoinPool threadPool =
+      new ForkJoinPool(Runtime.getRuntime().availableProcessors());
```

Comment:
1. `Runtime.getRuntime().availableProcessors()` is not needed, empty constructor is equivalent to this.
2. We probably should not use ForkJoinPool for this purpose, since event sources are IO bound operations. A normal executor service would be a good choice imo, maybe a configurable size; or a size that is same as event source number (but with an upper bound)
see:
https://www.geeksforgeeks.org/java-forkjoinpool-vs-executorservice/

Reply:
Good point, I originally was using the executor services defined by the `ConfigurationService` but wasn't seeing changes due to me being dumb (i.e. not using the proper version of the SDK in my testing code) and then forgot to go back to it when I finally managed to get it to work as I was expecting. 😅